crates/dekaf/src/session.rs

```diff
@@ -1103,13 +1104,21 @@ impl Session {
     #[instrument(skip_all, fields(group=?req.group_id))]
     pub async fn offset_commit(
         &mut self,
-        req: messages::OffsetCommitRequest,
+        mut req: messages::OffsetCommitRequest,
         header: RequestHeader,
     ) -> anyhow::Result<messages::OffsetCommitResponse> {
-        let mut mutated_req = req.clone();
-        for topic in &mut mutated_req.topics {
+        let collection_partitions = self
+            .fetch_collection_partitions(req.topics.iter().map(|topic| &topic.name))
```

Comment:
I noticed that `fetch_collection_partitions` calls `Collection::new`, which is called again on line 1149. And also that `Collection::new` is making a number of API requests. Seems worth trying to avoid the extra requests, if possible.

Also just a nit that you can feel free to ignore, but I personally did not expect a function called `new` to be making network requests. Might be worth renaming and/or refactoring `Collection::new` at some point.

Reply:
Refactored this to be `fetch_collections` and then map those into their partition counts etc where it's called